### PR TITLE
Thief roony primo

### PR DESCRIPTION
**Pull request by BraveMole**
Roony primo can steal an item for 3 seconds, the item is then dropped on the ground

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/castes/runner/castedatum_runner.dm [hunk 1]
```diff
@@ -151,3 +150,46 @@
 
 	// *** Ranged Attack *** //
 	pounce_delay = 13 SECONDS
+
+/datum/xeno_caste/runner/primordial
+	upgrade_name = "Primordial"
+	caste_desc = "Not what you want to run into in a dark alley. It looks extremely deadly."
+	primordial_message = "We are the fastest assassin of all time. Our speed is unmatched."
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/runner/castedatum_runner.dm:157]: Runner
Desc: "A sprinting terror of the hive. It looks ancient and menacing."
Message: "Nothing can outrun us. We are the swift death."

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by TiviPlus** [comment]
Also the icon should be ne dir not 4

File: code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm [hunk 2]
```diff
@@ -127,7 +127,7 @@
 		else
 			to_chat(X, span_xenodanger("We attempt to savage our victim, but we aren't yet ready."))
 
-	playsound(X.loc, prob(95) ? 'sound/voice/alien_pounce.ogg' : 'sound/voice/alien_pounce2.ogg', 25, TRUE)
+	playsound(X.loc, 'sound/voice/alien_pounce.ogg', 25, TRUE)
```
[the remark this answers is not in the document]
Comment (BraveMole) [reply code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm:131]: i'm using this sound on the snatch, having both sound on two different abilities would be confusing for the marine

File: code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm [hunk 3]
```diff
@@ -397,3 +397,81 @@
...
 
+/datum/action/xeno_action/activable/snatch
+	name = "Snatch"
+	action_icon_state = "evasion"
+	mechanics_text = "Take an item equipped by your target in your mouth, and carry it away."
+	plasma_cost = 75
+	cooldown_timer = 60 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_SNATCH
+	///Mutable appearance of the stolen item
+	var/mutable_appearance/stolen_appearance
```
[the remark this answers is not in the document]
Comment (BraveMole) [reply code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm:408]: The target no longer has the item on a slot, why ? You mean the xenomorph?

File: code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm [hunk 4]
```diff
@@ -397,3 +397,81 @@
...
+	var/obj/item/stolen_item = victim.get_active_held_item()
+	if(!stolen_item)
+		stolen_item = victim.get_inactive_held_item()
+		if(!stolen_item)
+			stolen_item = victim.get_item_by_slot(SLOT_S_STORE)
+			if(!stolen_item)
+				stolen_item = victim.get_item_by_slot(SLOT_BACK)
+				if(!stolen_item)
+					stolen_item = victim.get_item_by_slot(SLOT_SHOES)
+	if(!stolen_item)
```
[the remark this answers is not in the document]
Comment (BraveMole) [reply code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm:433]: if it were only slots, yes, but i need to take the active hand then inactive hand. So i would have to use callbacks?

Don't think a loop is worth

File: code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm (hunk 3 above)
[the remark this answers is not in the document]
Comment (TiviPlus) [reply code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm:408]: ahh sorry forgot to say that i meant the xenoability target flags

File: code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm (hunk 4 above)
[the remark this answers is not in the document]
Comment (TiviPlus) [reply code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm:433]: you can do the items and then loop through a list of slots after

File: code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm [hunk 5]
```diff
@@ -397,3 +397,89 @@
 		A = new /obj/effect/temp_visual/xenomorph/afterimage(T, owner) //Create the after image.
 		A.pixel_x = pick(rand(X.pixel_x * 3, X.pixel_x * 1.5), rand(0, X.pixel_x * -1)) //Variation on the X position
 
+/datum/action/xeno_action/activable/snatch
+	name = "Snatch"
+	action_icon_state = "evasion"
+	mechanics_text = "Take an item equipped by your target in your mouth, and carry it away."
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm:403]: needs unique

File: code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm [hunk 6]
```diff
@@ -397,3 +397,89 @@
...
+	. = ..()
+	if(!.)
+		return
+	if(!ishuman(A))
+		to_chat(owner, span_xenowarning("You cannot steal from that target"))
+		return FALSE
+	succeed_activate()
+	if(!do_after(owner, 0.5 SECONDS, FALSE, A, BUSY_ICON_HOSTILE))
+		to_chat(owner, span_xenowarning("Your victim moved, you failed to snatch an item"))
+		return FALSE
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm:427]: this should be in use ability for consistency

**Review by TiviPlus — CHANGES_REQUESTED** [review]
rest looks ok

File: code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm [hunk 7]
```diff
@@ -397,3 +397,89 @@
...
+	addtimer(CALLBACK(src, .proc/drop_item, stolen_item), 3 SECONDS)
+	RegisterSignal(owner, COMSIG_ATOM_DIR_CHANGE, .proc/owner_turned)
+	owner_turned(null, null, owner.dir)
+	add_cooldown()
+
+///Signal handler to update the item overlay when the owner is changing dir
+/datum/action/xeno_action/activable/snatch/proc/owner_turned(datum/source, old_dir, new_dir)
+	SIGNAL_HANDLER
+	if(!new_dir || new_dir == old_dir)
+		return
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm:455]:
```suggestion
```
new dir should never be null anyway and old dir/new dir is checked at the signal

File: code/modules/mob/living/carbon/xenomorph/hive_upgrades.dm [hunk 8]
```diff
@@ -291,3 +291,9 @@ GLOBAL_LIST_INIT(upgrade_categories, list("Buildings", "Defences", "Xenos"))//,
 	desc = "Unlocks the primordial defenders centrifugal force. An ability that allows them to rapidly spin and attack enemies nearby."
 	psypoint_cost = 75
 	icon = "primodefender"
+
+/datum/hive_upgrade/primordial/runner
+	name = PRIMORDIAL_DEFENDER
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/hive_upgrades.dm:296]: yes.... defender

File: code/modules/mob/living/carbon/xenomorph/castes/runner/castedatum_runner.dm [hunk 9]
```diff
@@ -126,7 +126,6 @@
 	caste_desc = "Not what you want to run into in a dark alley. It looks extremely deadly."
 	ancient_message = "We are the fastest assassin of all time. Our speed is unmatched."
 	upgrade = XENO_UPGRADE_THREE
-	wound_type = "runner" //used to match appropriate wound overlays
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/runner/castedatum_runner.dm:129]: no primo upgrade name added

Comment (TiviPlus) [reply code/modules/mob/living/carbon/xenomorph/castes/runner/castedatum_runner.dm:129]: also why remove this?

**Review by TiviPlus — COMMENTED** [review]
rest lgtm iassume you still need ability sprite

**Comment by BraveMole** [comment]
> rest lgtm iassume you still need ability sprite

no it's done and added

File: code/modules/mob/living/carbon/xenomorph/castes/runner/castedatum_runner.dm (hunk 9 above)
Comment (BraveMole) [reply code/modules/mob/living/carbon/xenomorph/castes/runner/castedatum_runner.dm:129]: uneeded, it's on the parent

File: code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm (hunk 7 above)
Comment (BraveMole) [reply code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm:455]: nop

File: code/modules/mob/living/carbon/xenomorph/castes/wraith/castedatum_wraith.dm [hunk 10]
```diff
@@ -134,7 +134,8 @@
 /datum/xeno_caste/wraith/primordial
 	upgrade_name = "Primordial"
 	caste_desc = "A xenomorph that has perfected the manipulation of space and time. Its movements appear quick and distorted."
-	primordial_message = "We control the very fabric of reality." //Help
+	primordial_message = "We control the very fabric of reality."
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/wraith/castedatum_wraith.dm:137]:
```suggestion
	primordial_message = "Mastery is achieved when \'telling time\' becomes \'telling time what to do\'."
```

File: code/modules/mob/living/carbon/xenomorph/castes/runner/castedatum_runner.dm [hunk 11]
```diff
@@ -151,3 +150,47 @@
 
 	// *** Ranged Attack *** //
 	pounce_delay = 13 SECONDS
+
+/datum/xeno_caste/runner/primordial
+	upgrade_name = "Primordial"
+	caste_desc = "A sprinting terror of the hive. It looks ancient and menacing."
+	primordial_message = "Nothing can outrun us. We are the swift death."
+	primordial_upgrade_name = PRIMORDIAL_RUNNER
+	upgrade = XENO_UPGRADE_FOUR
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/runner/castedatum_runner.dm:159]: upgrade names go on root

File: code/modules/mob/living/carbon/xenomorph/castes/wraith/castedatum_wraith.dm (hunk 10 above)
Comment (BraveMole) [reply code/modules/mob/living/carbon/xenomorph/castes/wraith/castedatum_wraith.dm:137]: we don't accept cringe here

**Comment by TiviPlus** [comment]
Also white pixel in ability ico n

File: code/modules/mob/living/carbon/xenomorph/castes/wraith/castedatum_wraith.dm (hunk 10 above)
Comment (TiviPlus) [reply code/modules/mob/living/carbon/xenomorph/castes/wraith/castedatum_wraith.dm:137]: Wrong

Comment (TiviPlus) [reply code/modules/mob/living/carbon/xenomorph/castes/wraith/castedatum_wraith.dm:137]: u r cring

File: code/modules/mob/living/carbon/xenomorph/hive_upgrades.dm [hunk 12]
```diff
@@ -292,6 +292,17 @@ GLOBAL_LIST_INIT(upgrade_categories, list("Buildings", "Defences", "Xenos"))//,
...
+	desc = "Unlocks the primordial runner snatch ability. An ability that allows them to steal equipped items momentarily."
+	psypoint_cost = 75
+	icon = "primorunner"
+
+/datum/hive_upgrade/primordial/wraith
+	name = PRIMORDIAL_WRAITH
+	desc = "Unlocks the primordial runner snatch ability. An ability that allows them to steal equipped items momentarily."
+	psypoint_cost = 125
+	icon = "primorunner"
 /datum/hive_upgrade/primordial/hivelord
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/hive_upgrades.dm:306]:
```suggestion
	psypoint_cost = 125
	icon = "primorunner"

/datum/hive_upgrade/primordial/hivelord
```

**Pull request merged**